src/interaction/command/circle.ts

```diff
@@ -0,0 +1,212 @@
+        option.setRequired(true);
+        return option;
+      });
+      subcommand.addStringOption((option) => {
+        option.setName("graphic");
+        option.setDescription("The graphic of the circle (url)");
+        option.setRequired(true);
+        return option;
+      });
+      subcommand.addUserOption((option) => {
+        option.setName("owner");
+        option.setDescription("The owner of the circle (mention them)");
+        option.setRequired(true);
+        return option;
+      });
+      return subcommand;
+    });
+  }
+
+  protected buildSlashCommand() {}
+
+  public async handleInteraction({ bot, interaction }: SlashCommandContext) {
+    const { member, guild } = interaction;
+    const subComm = interaction.options.getSubcommand();
+    switch (subComm) {
+      case "add":
+        await interaction.deferReply();
+        await addCircle(bot, interaction);
+    }
+  }
+}
+
+async function addCircle(bot: Bot, interaction: CommandInteraction) {
+  // const wizard = new Wizard(msg, undefined, {
+  //   title: "__**Circle Creation**__ ",
+  // });
+  // wizard.addNodes([
+  //   new UserMentionWizardNode(wizard, {
+  //     title: "Owner",
+  //     description: `Who's the owner of the circle? (mention them)`,
+  //   }),
+  //   new TextWizardNode(wizard, {
+  //     title: "Name",
+  //     description: `What's the circle name?`,
+  //   }),
+  //   new TextWizardNode(wizard, {
+  //     title: "Description",
+  //     description: `What's the circle description?`,
+  //   }),
+  //   new ColorWizardNode(wizard, {
+  //     title: "Color",
+  //     description: `What's the circle color? (used for the embed & role)`,
+  //   }),
+  //   new EmojiWizardNode(wizard, {
+  //     title: "Emoji",
+  //     description: `What's the circle's emoji?`,
+  //   }),
+  //   new GraphicWizardNode(wizard, {
+  //     title: "Image",
+  //     description: `What's the circle's graphic/image? (url)`,
+  //   }),
+  // ]);
+  // const res = await wizard.start();
+  // if (res === false) return;
+  const circle: CircleData = {
+    name: interaction.options.getString("name", true),
+    description: interaction.options.getString("description", true),
+    emoji: interaction.options.getString("emoji", true),
+    imageUrl: interaction.options.getString("graphic", true),
+    createdOn: new Date(),
+    owner: interaction.options.getUser("owner", true).id,
+    leaders: [],
+    subChannels: [],
+  };
+  const owner = await bot.users.fetch(circle.owner!);
+
+  if (owner.bot) {
+    await interaction.editReply("The owner cannot be a bot.");
+    return;
+  }
+
+  const color = interaction.options.getString("color", true) as ColorResolvable;
+
+  const guild = interaction.guild!;
+  const circleRole = await guild.roles.create({
+    name: `${circle.emoji} ${circle.name}`,
+    mentionable: true,
+    color,
+  });
+  const ownerMember = await guild.members.fetch(circle.owner!);
```

Comment:
can use `interaction.options.getMember` instead of `getUser` above, and just do everything using that member object

Reply:
Didn't know that was a function, thanks for the heads up